### PR TITLE
Create default yaml for s390x zVM and adapt test suites to use it in  YaST group

### PR DESCRIPTION
**Pull request by hjluo**
We need to create a default YAML for s390x zVM and adapt test suites to use it in the
YaST group.

- Ticket:
   https://progress.opensuse.org/issues/119839

- Needles:
  N/A

- VR:
    [s390x_zvm_default_test_result](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=hjluo%2Fos-autoinst-distri-opensuse%23zVM_default)

- Related MR:
  https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/436

### CODE REVIEW

File: schedule/yast/sle/flows/default_s390x_zvm.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  - installation/clock_and_timezone/accept_timezone_configuration
+local_user:
+  - installation/authentication/use_same_password_for_root
+  - installation/authentication/default_user_simple_pwd
+software: []
+booting:
+  - installation/bootloader_settings/disable_boot_menu_timeout
+security: []
+default_systemd_target: []
+installation_settings:
```
Comment (jknphy) [inline schedule/yast/sle/flows/default_s390x_zvm.yaml:39]: `validate_ssh_service_enabled` & `open_ssh_port` should be under `security`

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml [hunk 2]
```diff
@@ -15,40 +15,24 @@ vars:
...
+  registration:
+    - installation/registration/skip_registration
+  extension_module_selection:
+    - installation/module_selection/select_nonconflicting_modules
+  add_on_product:
+    - installation/add_on_product_installation/accept_add_on_installation
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  software:
+    - installation/select_patterns
```
Comment (jknphy) [inline schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:27]: recently we have: `installation/select_visible_patterns_from_bottom`

File: schedule/yast/sle/flows/default_s390x_zvm.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+registration:
+  - installation/registration/register_via_scc
+extension_module_selection:
+  - installation/module_registration/skip_module_registration
+add_on_product:
+  - installation/add_on_product/skip_install_addons
+add_on_product_installation: []
+system_role:
+  - installation/system_role/accept_selected_role_text_mode
+guided_filesystem: []
```
Comment (jknphy) [inline schedule/yast/sle/flows/default_s390x_zvm.yaml:26]: s/guided_filesystem/guided_partitioning/
let's use a generic hook name here, we will see later how it goes...

File: schedule/yast/ssh-x@s390x_zvm.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  Conduct an installation using ssh with X-Forwarding.
+  Might only be effective for zVM and powerVM
+vars:
+  YUI_REST_API: 1
+  DESKTOP: textmode
+  VIDEOMODE: ssh-x
+schedule:
+  access_beta:
+    - installation/access_beta_distribution
+    - installation/ensure_installer_fullscreen
```
Comment (jknphy) [inline schedule/yast/ssh-x@s390x_zvm.yaml:12]: this is arguable, but the resize seems to be needed from the selection screen, to not mix thing, we could do instead:
```
product_selection:
  - installation/ensure_installer_fullscreen
  - installation/product_selection/install_SLES
```

File: schedule/yast/ssh-x@s390x_zvm.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  guided_filesystem:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/accept_default_part_scheme
+    - installation/partitioning/guided_setup/accept_default_fs_options
```
Comment (jknphy) [inline schedule/yast/ssh-x@s390x_zvm.yaml:16]: missing:
```
default_systemd_target:
    - installation/installation_settings/validate_default_target
```

File: schedule/yast/ssh-x@s390x_zvm.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+name:           ssh-X
```
Comment (jknphy) [inline schedule/yast/ssh-x@s390x_zvm.yaml:1]: let's create a folder for those splits so we have: `/yast/ssh-x/ssh-x@s390x_zvm.yaml`

**Review by jknphy — COMMENTED** [review]
Looks pretty good! some minor catches.

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml [hunk 7]
```diff
@@ -15,40 +15,24 @@ vars:
   ADDONS: all-packages
```
Comment (jknphy) [inline schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:15]: let's make consistent the name of the file using `...s390x_zvm` all lower letters and with underscore, could you please apply to the rest of the file and MR?

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml (hunk 2 above)
Comment (hjluo) [reply schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:27]: changed.

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml (hunk 7 above)
Comment (hjluo) [reply schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:15]: OK

File: schedule/yast/ssh-x@s390x_zvm.yaml (hunk 5 above)
Comment (hjluo) [reply schedule/yast/ssh-x@s390x_zvm.yaml:16]: added.

File: schedule/yast/ssh-x@s390x_zvm.yaml (hunk 4 above)
Comment (hjluo) [reply schedule/yast/ssh-x@s390x_zvm.yaml:12]: changed.

File: schedule/yast/ssh-x@s390x_zvm.yaml (hunk 6 above)
Comment (hjluo) [reply schedule/yast/ssh-x@s390x_zvm.yaml:1]: OK

File: schedule/yast/sle/flows/default_s390x_zvm.yaml (hunk 1 above)
Comment (hjluo) [reply schedule/yast/sle/flows/default_s390x_zvm.yaml:39]: changed.

File: schedule/yast/sle/flows/default_s390x_zvm.yaml (hunk 3 above)
Comment (hjluo) [reply schedule/yast/sle/flows/default_s390x_zvm.yaml:26]: changed.

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml (hunk 2 above)
Comment (hjluo) [reply schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:27]: changed.

File: schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml (hunk 7 above)
Comment (hjluo) [reply schedule/yast/select_modules_and_patterns+registration/select_modules_and_patterns+registration@s390x-zVM.yaml:15]: OK

File: schedule/yast/sle/guided_ext4/guided_ext4_s390x_zvm.yaml [hunk 8]
```diff
@@ -6,7 +6,9 @@ vars:
   FILESYSTEM: ext4
   YUI_REST_API: 1
 schedule:
-  guided_filesystem:
+  guided_partition:
```
Comment (jknphy) [inline schedule/yast/sle/guided_ext4/guided_ext4_s390x_zvm.yaml:9]: guided_partitioning

File: schedule/yast/sle/guided_xfs/guided_xfs_s390x_zvm.yaml [hunk 9]
```diff
@@ -5,7 +5,13 @@ description:    >
 vars:
   YUI_REST_API: 1
 schedule:
-  guided_filesystem:
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  guided_partition:
```
Comment (jknphy) [inline schedule/yast/sle/guided_xfs/guided_xfs_s390x_zvm.yaml:12]: guided_partitioning

File: schedule/yast/ssh-x/ssh-x@s390x_zvm.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  Might only be effective for zVM and powerVM
+vars:
+  YUI_REST_API: 1
+  DESKTOP: textmode
+  VIDEOMODE: ssh-x
+schedule:
+  product_selection:
+    - installation/ensure_installer_fullscreen
+    - installation/product_selection/install_SLES
+  guided_partition:
```
Comment (jknphy) [inline schedule/yast/ssh-x/ssh-x@s390x_zvm.yaml:13]: guided_partitioning

File: schedule/yast/minimal+base/minimal+base@yast-s390x-disk-activation.yaml [hunk 11]
```diff
@@ -11,45 +11,22 @@ vars:
...
-  - console/installation_snapshots
-  - console/zypper_lr
-  - console/zypper_ref
-  - console/ncurses
-  - update/zypper_up
-  - console/zypper_lifecycle
-  - console/orphaned_packages_check
-  - console/validate_installed_patterns
-  - console/consoletest_finish
+  software:
```
Comment (jknphy) [inline schedule/yast/minimal+base/minimal+base@yast-s390x-disk-activation.yaml:14]: let's modify the name of the file as well here, we had a weird name from before.

File: schedule/yast/ssh-x/ssh-x@s390x_zvm.yaml (hunk 10 above)
Comment (hjluo) [reply schedule/yast/ssh-x/ssh-x@s390x_zvm.yaml:13]: changed.

File: schedule/yast/minimal+base/minimal+base@yast-s390x-disk-activation.yaml (hunk 11 above)
Comment (hjluo) [reply schedule/yast/minimal+base/minimal+base@yast-s390x-disk-activation.yaml:14]: how about s/software/patterns/ ? here

File: schedule/yast/minimal+base/minimal+base@s390x_zvm.yaml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+---
+name: minimal+base@yast-s390x-disk-activation
```
Comment (jknphy) [inline schedule/yast/minimal+base/minimal+base@s390x_zvm.yaml:2]: also here.

**Comment by jknphy** [comment]
LGTM

**Pull request merged**